src/main/Database/Database.java

```diff
@@ -49,7 +51,9 @@ public void connect() {
             e.printStackTrace();
         }
     }
-    public void closeConnection(){
+
+    @Override
+    public void close(){
```

Comment:
this method is never called, best idea is to create new window listener that will execute on windowClosing or windowClosed